### PR TITLE
docs(samples): Add & update samples for STT v2

### PR DESCRIPTION
**Pull request by nitsanshai**
## Description

Added samples for:
* Changing STT endpoint location
* Automatic punctuation
* Profanity filter
* Selecting a model
* Multi-channel
* Word level confidence
* Word time offsets
* Chirp

Updated existing samples:
* Added commandline args
* Removed recognizer creation where not needed

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: speech/snippets/adaptation_v2_custom_class_reference.py [hunk 1]
```diff
@@ -13,15 +13,15 @@
 # limitations under the License.
 
 
-# [START speech_adaptation_v2_custom_class_reference]
```
[the remark this answers is not in the document]
Comment (nicain) [reply speech/snippets/adaptation_v2_custom_class_reference.py:16]: Ah nevermind, its not used in the sample, but in __main__. LG!

[the remark this answers is not in the document]
Comment (nitsanshai) [reply speech/snippets/adaptation_v2_custom_class_reference.py:16]: Even if the import isn't used in the region? it's only used outside of the region. But happy to move it

File: speech/snippets/transcribe_gcs_v2_test.py [hunk 2]
```diff
@@ -14,37 +14,23 @@
...
-
-
 @Retry()
 def test_transcribe_gcs_v2(capsys: pytest.CaptureFixture) -> None:
     project_id = os.getenv("GOOGLE_CLOUD_PROJECT")
 
-    recognizer_id = "recognizer-" + str(uuid4())
     response = transcribe_gcs_v2.transcribe_gcs_v2(
-        project_id, recognizer_id, "gs://cloud-samples-data/speech/audio.flac"
+        project_id, "gs://cloud-samples-data/speech/audio.flac"
```
Comment (nicain) [inline speech/snippets/transcribe_gcs_v2_test.py:29]: Consider making this a module_level variable: _TEST_AUDIO_FILE_PATH = "gs://..."

File: speech/snippets/transcribe_word_level_confidence_v2.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    )
+
+    # Transcribes the audio into text
+    response = client.recognize(request=request)
+
+    for result in response.results:
+        print(f"Transcript: {result.alternatives[0].transcript}")
+
+    return response
+
```
Comment (nicain) [inline speech/snippets/transcribe_word_level_confidence_v2.py:57]: Trim out this whitespace to save real estate on the page that shows this sample

File: speech/snippets/transcribe_word_time_offsets_v2.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    # Transcribes the audio into text
+    response = client.recognize(request=request)
+
+    for result in response.results:
+        print(f"Transcript: {result.alternatives[0].transcript}")
+
+    return response
+
+
```
Comment (nicain) [inline speech/snippets/transcribe_word_time_offsets_v2.py:58]: Same, remove whitespace

File: speech/snippets/transcribe_streaming_v2_test.py [hunk 5]
```diff
@@ -14,31 +14,21 @@
...
-from uuid import uuid4
 
 from google.api_core.retry import Retry
-from google.cloud.speech_v2 import SpeechClient
-from google.cloud.speech_v2.types import cloud_speech
 import pytest
 
 import transcribe_streaming_v2
 
 RESOURCES = os.path.join(os.path.dirname(__file__), "resources")
```
[the remark this answers is not in the document]
Comment (nitsanshai) [reply speech/snippets/transcribe_streaming_v2_test.py:23]: done!

File: speech/snippets/transcribe_streaming_voice_activity_timeouts.py [hunk 6]
```diff
@@ -71,7 +57,9 @@ def transcribe_streaming_voice_activity_timeouts(
         cloud_speech.StreamingRecognizeRequest(audio=audio) for audio in stream
     )
 
-    recognition_config = cloud_speech.RecognitionConfig(auto_decoding_config={})
+    recognition_config = cloud_speech.RecognitionConfig(
+        auto_decoding_config={}, language_codes=["en-US"], model="latest_long"
```
[the remark this answers is not in the document]
Comment (nitsanshai) [reply speech/snippets/transcribe_streaming_voice_activity_timeouts.py:61]: No, it needs to be not None, but I updated it be explicitly the empty proto message

**Review by balajismaniam — APPROVED** [review]

File: speech/snippets/transcribe_word_level_confidence_v2.py (hunk 3 above)
Comment (nitsanshai) [reply speech/snippets/transcribe_word_level_confidence_v2.py:57]: The linter/formatter commands actually add this space so I'm going to leave it :(

**Pull request merged**